cluster/kind/kind_test.go

```diff
@@ -0,0 +1,373 @@
+	}, {
+		desc: "failed to get name",
+		regs: []string{"us-west1-docker.pkg.dev"},
+		resp: []fexec.Response{
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "--password-stdin", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "kubectl", Args: []string{"config", "current-context"}, Err: "failed to get name"},
+		},
+		wantErr: "failed to get name",
+	}, {
+		desc: "failed to get nodes",
+		regs: []string{"us-west1-docker.pkg.dev"},
+		resp: []fexec.Response{
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "--password-stdin", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "kubectl", Args: []string{"config", "current-context"}, Stdout: "kind-kne"},
+			{Cmd: "kind", Args: []string{"get", "nodes", "--name", "kne"}, Err: "failed to get nodes"},
+		},
+		wantErr: "failed to get nodes",
+	}, {
+		desc: "failed to cp config to node",
+		regs: []string{"us-west1-docker.pkg.dev"},
+		resp: []fexec.Response{
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "--password-stdin", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "kubectl", Args: []string{"config", "current-context"}, Stdout: "kind-kne"},
+			{Cmd: "kind", Args: []string{"get", "nodes", "--name", "kne"}, Stdout: "kne-control-plane"},
+			{Cmd: "docker", Args: []string{"cp", ".*/config.json", "kne-control-plane:/var/lib/kubelet/config.json"}, Err: "failed to cp config to node"},
+		},
+		wantErr: "failed to cp config to node",
+	}, {
+		desc: "failed to restart kubelet",
+		regs: []string{"us-west1-docker.pkg.dev"},
+		resp: []fexec.Response{
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "--password-stdin", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "kubectl", Args: []string{"config", "current-context"}, Stdout: "kind-kne"},
+			{Cmd: "kind", Args: []string{"get", "nodes", "--name", "kne"}, Stdout: "kne-control-plane"},
+			{Cmd: "docker", Args: []string{"cp", ".*/config.json", "kne-control-plane:/var/lib/kubelet/config.json"}},
+			{Cmd: "docker", Args: []string{"exec", "kne-control-plane", "systemctl", "restart", "kubelet.service"}, Err: "failed to restart kubelet"},
+		},
+		wantErr: "failed to restart kubelet",
+	}}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			fexec.LogCommand = func(s string) {
+				t.Logf("%s: %s", tt.desc, s)
+			}
+			cmds := fexec.Commands(tt.resp)
+			kexec.Command = cmds.Command
+			defer checkCmds(t, cmds)
+
+			googleFindDefaultCredentials = func(_ context.Context, _ ...string) (*google.Credentials, error) {
+				if tt.findCredsErr {
+					return nil, errors.New("unable to find default credentials")
+				}
+				return &google.Credentials{TokenSource: &fakeTokenSource{tokenErr: tt.tokenSourceErr}}, nil
+			}
+
+			err := SetupGARAccess(ctx, tt.regs)
+			if s := errdiff.Substring(err, tt.wantErr); s != "" {
+				t.Fatalf("unexpected error: %s", s)
+			}
+		})
+	}
+}
+
+type fakeTokenSource struct {
+	tokenErr bool
```

Comment:
tokenErr reads to me an error instead a bool.. should we make it "hasTokenErr"?

Reply:
done